Update accessibility doc with Semantics roles information (#11971)
flutter/website: src/content/ui/accessibility-and-internationalization/accessibility.md

{% ytEmbed 'bWbBgbmAdQs', 'Building Flutter apps with accessibility in mind' %}

## Enhancing Accessibility with Semantic Roles

### What are Semantic Roles?

> Semantic roles define the purpose of a UI element, helping screen readers and other assistive tools interpret and present your application effectively to users. For example, a role can indicate if a widget is a button, a link, a heading, a slider, or part of a table. 

Reviewer: Global comment: shorten line length to less than 80 characters. It's best to shorten based on semantic breaks.

While Flutter's standard widgets often provide these semantics automatically, a custom component without a clearly defined role can be incomprehensible to a screen reader user.

By assigning appropriate roles, you ensure that:

* Screen readers can announce the type and purpose of elements correctly.
* Users can navigate your application more effectively using assistive technologies.
* Your application adheres to web accessibility standards, improving usability.

> ### Using `SemanticsRole` in Flutter for Web

Reviewer: ```suggestion
### Using `SemanticsRole` in Flutter for web
```

Flutter provides the [`Semantics` widget][] with the [`SemanticsRole` enum][] to allow developers to assign specific roles to their widgets. When your Flutter web app is rendered, these Flutter-specific roles are translated into corresponding ARIA roles in the web page's HTML structure.

[`SemanticsRole` enum]: {{site.api}}/flutter/dart-ui/SemanticsRole.html

**1. Automatic Semantics from Standard Widgets:**

> Many standard Flutter widgets, like `TabBar`, `MenuAnchor`, and `Table`, automatically include semantic information, including their roles. Whenever possible, prefer using these standard widgets as they handle many accessibility aspects out-of-the-box.

Reviewer: Avoiding "..include...including.."

```suggestion
Many standard Flutter widgets, like `TabBar`, `MenuAnchor`, and `Table`,
automatically include semantic information along with their roles.
Whenever possible, prefer using these standard widgets
as they handle many accessibility aspects out-of-the-box.
```

> **2. Explicitly Adding or Overriding Roles:**

Reviewer: ```suggestion
**2. Explicitly adding or overriding roles:**
```

Google dev docs use sentence case for titles/anchor, not title case.
